docs/contributing.rst

```diff
@@ -2,12 +2,635 @@
+    the test script, are defined either under a base test class or in the very
+    beginning of the test script for being used by multiple unit test cases.
+
+  - Any group of testing functions dedicated to testing a particular
+    phenomenon (e.g. a specific edge case, data structure, etc.) is
+    implemented by a class, which inherits `TestCase` from Python's
+    `unittest` and likely the base test class implemented for the purpose
+    mentioned above.
+
+  - Assertions are used for testing various cases such as array comparison.
+
+  - Please see previously implemented test cases for reference of the
+    recommended testing approach
+
+.. note:: Our test suite that includes all the unit tests is executed automarically
+for PRs with the help of GitHub Actions workflows to ensure new code passes tests.
+Hence, please check `TODO`_ to make sure your PR tests are all passing before asking
+others to review your work.
+
+3.7.2.2. Docstrings
+~~~~~~~~~~~~~~~~~~~
+
+All Python functions must contain a `Google Style Python
+<https://github.com/google/styleguide/blob/gh-pages/pyguide.md>`_ *docstring* (i.e.
+triple quoted comment blocks). These docstrings are accessed from the Python interpreter
+whenever the user types::
+
+    help <FUNCTION_NAME>
+
+They are also automatically converted into web-accessible documentation available from
+the `UXarray documentation <https://uxarray.readthedocs.io/>`_.
+
+The docstrings must contain:
+
+1. A brief description of the functionality provided. What does this function do?
+
+2. If available, references to the algorithm or implementation employed
+
+3. A complete description of arguments and return values in Google format
+
+   * Please refer to the existing functions that already have this
+
+4. One or more very short usage examples that demonstrates how to invoke
+   the function, and possibly what to expect it to return
+
+   * No need for these examples to actually be executable
+
+   * If a usage example is longer than a handful of lines, a more complete
+     example may be created instead, referring to `TODO `_.
+
+3.7.2.3. Documentation
+~~~~~~~~~~~~~~~~~~~~~~
+
+As we mentioned a few times throughout this guide, Uxarray has a static `documentation
+<https://uxarray.readthedocs.io/>`_ :ref:`index` page that is being generated automatically from the
+repository's code structure. However, there needs to be some manual additions to the
+proper documentation index file(s) for the automation to work. The following index files
+are used for UXarray documentation (directories relative from the root):
+
+    `docs/user_api/index.rst`
+    `docs/internal_api/index.rst`
+
+which allows the following documentation to be autoamtically generated, respectively:
 
-Anyone can contribute to and participate in the Uxarray project
-at any levels of project development! We conduct all of our work
-in the open, and all of our work is Open Source Licensed.
+    :ref:`user_api/index`
+    :ref:`internal_api/index`
 
```

Comment:
Could be a good use for cards!

Reply:
Thanks for the suggestion! I had changed the structure of that part into a more paragraph way in the meantime. Please let me know if you still think cards might be applicable for that part.